File: PROYECTO/V1/Zarpas_Backend/src/main/java/com/gestion/zarpas_backend/modelo/Comentario.java
package com.gestion.zarpas_backend.modelo;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.*;
import lombok.*;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "comentario")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Comentario {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_comentario")
    private Long idComentario;

    @ManyToOne
    @JoinColumn(name = "id_usuario", nullable = false)
    @JsonBackReference("usuario-comentarios")
    private Usuario usuario;

    @ManyToOne
    @JoinColumn(name = "id_publicacion", nullable = false)
    @JsonBackReference("publicacion-comentarios")
    private Publicacion publicacion;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String texto;

    @Column(name = "fecha_creacion")
    private Timestamp fechaCreacion;

    @Column(name = "fecha_modificacion")
    private Timestamp fechaModificacion;

    // Relación con la entidad de unión ComentarioReaccion
    @OneToMany(mappedBy = "comentario", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @JsonManagedReference("comentario-comentarioReacciones")
    private List<ComentarioReaccion> comentarioReacciones = new ArrayList<>();

    @JsonProperty("usernameUsuario")
    public String getUsernameFromUsuario() {
        return (this.usuario != null) ? this.usuario.getUsername() : null;
    }
}

File: PROYECTO/V1/Zarpas_Frontend/gestion-zarpas-frontend/src/app/modelos/Comentario.ts
import { TipoReaccion } from "./TipoReaccion";
import { Usuario } from "./Usuario";


export interface Comentario {
    idComentario?: number;
    usuario: Usuario;
    usernameUsuario?: string;
    idPublicacion?: number;
    texto: string;
    fechaCreacion?: Date | null;
  fechaModificacion?: Date | null;
  reaccionesConteo?: { like: number; dislike: number };
  currentUserReaction?: TipoReaccion | null;

  conteoLikes?: number;    
  conteoDislikes?: number;
  }

export interface ComentarioRequest {
    idUsuario: number;
    idPublicacion: number;
    texto: string;
  }